src/Cli/dotnet/commands/dotnet-new/MSBuildEvaluation/CapabilityExpressionEvaluator.cs

```diff
@@ -0,0 +1,302 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using LocalizableStrings = Microsoft.DotNet.Tools.New.LocalizableStrings;
+
+namespace Microsoft.TemplateEngine.MSBuildEvaluation
+{
+    internal class CapabilityExpressionEvaluator
+    {
+        /// <summary>
+        /// The set of terms that are present.
+        /// </summary>
+        private readonly IReadOnlyList<string> _presentTerms;
+
+        /// <summary>
+        /// The tokenizer that reads the expression.
+        /// </summary>
+        private readonly Tokenizer _tokenizer;
+
+        /// <summary>
+        /// The set of disallowed characters in terms.
+        /// </summary>
+        /// <remarks>
+        /// We restrict many symbols, especially mathematical symbols, because we may eventually want to 
+        /// support arithmetic expressions.
+        /// </remarks>
+        internal static readonly char[] DisallowedCharacters = "\"'`:;,+-*/\\!~|&%$@^()={}[]<>? \t\b\n\r".ToCharArray();
```

Comment:
This seems too permissive (e.g. not all unicode whitespaces are restricted). How about justr restricting to alphanums? Is there any requirement about what needs to be supported?

Reply:
this matches VS implementation, so better keep as is.